cmake/Modules/FindSYCLToolkit.cmake

```diff
@@ -49,12 +49,27 @@ find_file(
   )
 
 # Find SYCL library fullname.
-find_library(
-  SYCL_LIBRARY
-  NAMES sycl
-  HINTS ${SYCL_LIBRARY_DIR}
-  NO_DEFAULT_PATH
-)
+# On both Linux and Windows, the current version of SYCL runtime is 7.
+# On Linux, libsycl.so symbolic links to libsycl.so.7.
+# On Windows, sycl7.dll version suffix is used.
+foreach(sycl_lib_version "" 7)
```

Comment:
Is it good to assume the version of sycl to `7` here? What's the problem of original implementation?

Reply:
It does not make sense to hard code a particular version here.

Reply:
The current version of SYCL runtime from oneAPI is `7`, so I've hardcoded to `7`. 

Problem with the original implementation is that there's no `sycl.dll` on Windows. Only `sycl7.dll` with version suffix. `sycl7.dll` on Windows corresponds to `libsycl.so` on Linux.
In the future, if compiler team adds support to `sycl.dll` on Windows, we can drop this. Otherwise, we need to update `sycl_lib_version` to `8` and so on when SYCL runtime version uplifts.

Reply:
We may need to request compiler team to support `sycl.{dll, lib}` without version suffix to avoid hardcoding and maintaining particular version. In the meantime, I don't think there's an alternative?